### ISSUE TITLE
Get rid of the WinUI 3 Windows Runtime Component template for C++

### ISSUE THREAD

**Issue by jaigak** [microsoft/WindowsAppSDK#1646]
Consider removing the WinUI 3 Windows Runtime Component template for C++ since one can just as easily use the old templates shipped in VS for C++/CX and the extension for C++/WinRT and reference WinUI 3 and WindowsAppSDK if they wanted to. There's too many templates in VS already and this just adds more bloat. The C# one could remain as it targets a different runtime altogether which makes it less of a good choice for most libraries as they would want to support all kinds of apps possible.

**Comment by Jaiganeshkumaran** [comment]
> But WinUI3's C++ templates just added WIL include and package reference, so technically they are not the same as those in C++/CX or C++/WinRT.

That's right but if the developers want it, they can add it manually. Unlike the app templates, the component doesn't really contain any code. There shouldn't be a reason for a duplicated template that has nothing but some NuGet references.

**Comment by sylveon** [comment]
The Windows Runtime Component template for WinUI 3 supports being compiled targetting the desktop API set, and uses the WinUI 3 XAML compiler instead of the system XAML compiler

**Comment by Jaiganeshkumaran** [comment]
> The Windows Runtime Component template for WinUI 3 supports being compiled targetting the desktop API set, and uses the WinUI 3 XAML compiler instead of the system XAML compiler

It could be implemented in a wizard-like fashion instead of being bloated up with too many similar templates.
